src/main/java/net/mcreator/plugin/Plugin.java

```diff
@@ -106,11 +106,12 @@ public int getWeight() {
 	/**
 	 * <p>The plugin is compatible when the version of MCreator used is included in the supported versions list.</p>
 	 * <p>When the supported versions list is null, the plugin is compatible with all versions of MCreator.</p>
+	 * <p>When the supported versions list contains the short version of MCreator, the plugin is compatible with that MCreator release.</p>
 	 *
 	 * @return <p>If the plugin is compatible with the version used.</p>
 	 */
 	public boolean isCompatible() {
-		return supportedversions == null || supportedversions.contains(Launcher.version.versionlong);
+		return supportedversions == null || supportedversions.contains(Launcher.version.versionlong) || (supportedversions.contains(Launcher.version.major) && !Launcher.version.isSnapshot());

```

Comment:
You are doing contains check in long array with String. I have high doubts this way implemented correctly

Reply:
I tried it before publishing the PR and it worked

Reply:
![image](https://github.com/MCreator/MCreator/assets/16374228/ede6d39b-404c-4139-8920-e76860bfcd50)

Reply:
What you are doing is undefined and even if it works, it is unexpected behaviour